index.js

```diff
@@ -33,6 +68,7 @@ async function loadVaultSecrets() {
         console.error('Path to secrets not specified but vault is enabled');
         throw new Error('Path to secrets not specified but vault is enabled');
```

Comment:
non-fatal errors in this function can be thrown, while possibly still leading to a successful request.

Reply:
Kinda curious why a vault error isn't assumed to be fatal, but an ego auth error is? or is the approach just that a vault error will likely lead to an auth error?

Reply:
the reason I chose that is that in lines 111-113 there's a set of fallbacks to `process.env` values... so there is a chance the function itself is not needed.
<img width="682" alt="image" src="https://user-images.githubusercontent.com/2107110/191343685-20a00459-5f6f-4df1-bdbc-fce6f83a58a1.png">

Worst case scenario, if the request fails afterwards, the process will exit in error, even if not immediately.

Reply:
cool cool...and we have logs so we'll still know the vault result 👍